scripts/partition_manager.py

```diff
@@ -493,33 +491,65 @@ def get_dynamic_area_start_and_size(static_config, flash_size):
     return start, end - start
 
 
-def get_pm_config(input_config, flash_start, flash_size, static_config):
-    to_resolve = dict()
-    start = flash_start
+def get_ram_config(pm_config, ram_start, ram_size, static_configuration=None):
```

Comment:
I believe the below code will be simpler if we do ={} here.

Generally, None should be avoided.

Specifically here we can omit the ternary operation because sum([]) == 0 and [a for a,b in {}.items()] == []

Reply:
pep8 disallows mutable default arguments, so I will stick with None, I have updated the check below to avoid the issue you highlighted

Reply:
See https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1879/commits/6e0b3281ce9f3570db793426114ae46530cdcdcf#diff-1306477692c2145a0182b10e022177c6R494

Reply:
Oh, I was not aware.

Reply:
> Generally, None should be avoided.

Just my two cents: None should not be avoided if the default value is mutable, and not just to avoid making the linter happy.

The default value of a keyword argument is only evaluated once, not each time the function is called, which results in this common gotcha:

```
$ cat ~/foo.py
import time

def foo(bar={}):
    bar[time.time()] = 'foo timestamp'
    return bar

print(foo())
print(foo())

$ python3 ~/foo.py
{1581522358.6238706: 'foo timestamp'}
{1581522358.6238706: 'foo timestamp', 1581522358.623899: 'foo timestamp'}
```